crates/stages/src/pipeline/mod.rs

```diff
@@ -245,14 +247,15 @@ where
         // Unwind stages in reverse order of execution
         let unwind_pipeline = self.stages.iter_mut().rev();
 
-        let mut tx = Transaction::new(&self.db)?;
+        let shareable_db = ShareableDatabase::new(&self.db, self.chain_spec.clone());
```

Comment:
Would like us to move to something like `trait Backend` instead of `trait Database` and call the struct `Database::new(backend)` or something like that. I don't like the "ShareableDatabase" name, doesn't really convey anything useful